### ISSUE TITLE
how to retrain my own image classifier using Alexnet model?

### ISSUE THREAD

**Issue by willduan** [tflearn/tflearn#723]
I use alexnet to train my own image classifier, the image structure is like this:
ROOT_FOLDER -> SUBFOLDER_0 (CLASS 0) -> CLASS0_IMG1.jpg 
									 -> CLASS0_IMG2.jpg -> ...
			-> SUBFOLDER_1 (CLASS 1) -> CLASS1_IMG1.jpg -> ...
									 -> ...
									 
and the labels are the subfolders' name. I just want to implement a a generic framework. The main problem now is that I don't know how many categories are there now. and i don't know the last layer's output. And the code is wong, how can I implement that.

`
from __future__ import division, print_function, absolute_import
from tflearn.layers.core import input_data, dropout, fully_connected
from tflearn.layers.conv import conv_2d, max_pool_2d
from tflearn.layers.normalization import local_response_normalization
from tflearn.layers.estimator import regression

from tflearn.data_utils import image_preloader, build_hdf5_image_dataset
import tflearn
import tensorflow as tf
import os

def model_main():

    # Building alexNet
    network = input_data(shape = [None, 227, 227, 3])
    network = conv_2d(network, 96, 11, strides = 4, activation = 'relu')
    network = max_pool_2d(network, 3, strides = 2) #kernel_size=3

    network = local_response_tflearn.config.init_graph (seed=None, log_device=False, num_cores=0, gpu_memory_fraction=0, soft_placement=True)

normalization(network)
    network = conv_2d(network, 256, 5, activation = 'relu')
    network = max_pool_2d(network, 3, strides=2)
    network = local_response_normalization(network)

    network = conv_2d(network, 384, 3, activation='relu')
    network = conv_2d(network, 384, 3, activation='relu')
    network = conv_2d(network, 256, 3, activation='relu')
    network = max_pool_2d(network, 3, strides=2)
    network = local_response_normalization(network)

    network = fully_connected(network, 4096, activation = 'tanh')
    network = dropout(network, 0.5)

    network = fully_connected(network, 4096, activation = 'tanh')
    network = dropout(network, 0.5)
   
    network = fully_connected(network, 1, activation = 'softmax') #or softmax
    
    network = tflearn.reshape(network, new_shape = [-1])
    
    network = regression(network, optimizer = 'momentum', loss = 'categorical_crossentropy', learning_rate = 0.001)
    
    return network

def train(network):
    X, Y = image_preloader('./flower_photos/', image_shape = (227, 227), mode = 'folder', categorical_labels = False, normalize = True)

    print(type(X))
    print(type(Y))
    # Training
    tflearn.config.init_graph(gpu_memory_fraction = 0.9)
    model = tflearn.DNN(network, checkpoint_path = './model_alexnet', max_checkpoints = 1, tensorboard_verbose = 2)
    model.fit(X, Y, n_epoch = 10, validation_set = 0.1, shuffle = True,
                show_metric = True, batch_size = 32, snapshot_step = 200,
                        snapshot_epoch = False, run_id='alexnet')
    model.save('flower-classifier')

if __name__ == "__main__":
    model = model_main()
    train(model)

`

I don't know how to rewrite the last layer, thank you!

**Comment by aymericdamien** [comment]
You need to set `categorical_labels = True`, and indicate how many class you have for the latest fully connected layer.

**Comment by willduan** [comment]
@aymericdamien thanks very much!

**Comment by willduan** [comment]
I run the AlexNet and i found that the model's predict output just numbers. And my dataset is flower_photos, the folder is:
ROOT_FOLDER -> daisy -> CLASS0_IMG1.jpg
                                         -> CLASS0_IMG2.jpg
                                         -> ...
                            -> roses -> CLASS1_IMG1.jpg
                                         -> ...
                               -> ...

and when I predict an image, it output the number label, is there any method exists to output the real label? Becase I don't know the correspondence of X and Y when use  ```image_preloader```.
